test/e2e/tests/reset-account.spec.js

```diff
@@ -0,0 +1,68 @@
+const { strict: assert } = require('assert');
+const { convertToHexValue, withFixtures } = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('Reset account', function () {
+  // ////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  // When user get stuck with pending transactions, one can reset the account by clicking the 'Reset account' //
+  // button in settings, advanced tab. This functionality will clear all the send transactions history.       //
+  // Note that the receive transactions history will be kept and it only only affects the current network.    //
+  // ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+  it('User can reset account via the advanced setting tab, ', async function () {
+    const ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+          balance: convertToHexValue(25000000000000000000),
+        },
+      ],
+    };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // Send ether
```

Comment:
Here we could use one of the existing fixtures, like `withTransactionControllerMultipleTransactions` so we can skip the `Send ether` and `Check completed send transaction is displayed` steps, and jump directly to Reset account

Reply:
I replaced `send ether` section by `withTransactionControllerMultipleTransactions` fixture. Thanks ! 
I kept the check that send history is displayed so i can compare after reset account we don't have send history, which makes the test more readable and don't cost more execution time. Hope it works for you.

Reply:
https://github.com/MetaMask/metamask-extension/pull/16996#discussion_r1053598713 Maybe we can create a new method `withTransactionControllerConfirmedTransactions` or similar

Reply:
Updated to use existing `withTransactionControllerApprovedTransaction()` controller, please check, thanks.

Reply:
Added a new method `withTransactionControllerCompletedTransaction`, please review, thanks.